### PR TITLE
V8 test generic

### PR DESCRIPTION
**Pull request by bluebill1049**

### CODE REVIEW

**Comment by bluebill1049** [comment]
cc @Moshyfawn on the generic usage in v8.

**Comment by Moshyfawn** [comment]
Thx, gonna check it out after work

File: src/__tests__/type.test.tsx [hunk 1]
```diff
@@ -171,3 +181,66 @@ test('should infer context type into control', () => {
...
+
+  const Input = <T extends FieldValues, P extends PathString>(
+    props: InputProps<T, P>,
+    ref: React.Ref<HTMLInputElement>,
+  ) => {
+    const { field } = useController({
+      name: of(props.name),
+      control: props.control,
+    });
+    const [value, setValue] = React.useState(field.value || '');
```
Comment (bluebill1049) [inline src/__tests__/type.test.tsx:208]: correct type inference for the input value.

File: src/__tests__/type.test.tsx [hunk 2]
```diff
@@ -171,3 +181,66 @@ test('should infer context type into control', () => {
...
+      firstName: string;
+      lastName: string;
+    };
+    age: string;
+    pet: { name: string }[];
+  };
+
+  type InputProps<T extends FieldValues, P extends PathString> = {
+    control: Control<T>;
+    name: Auto.TypedFieldPath<T, P, string>;
```
Comment (bluebill1049) [inline src/__tests__/type.test.tsx:197]: Correct type check for field name (path)

File: src/__tests__/type.test.tsx [hunk 3]
```diff
@@ -171,3 +181,66 @@ test('should infer context type into control', () => {
...
+          <Input name="age" control={control} />
+
+          <input type="submit" />
+        </form>
+      </div>
+    );
+  }
+
+  App;
+});
```
Comment (bluebill1049) [inline src/__tests__/type.test.tsx:246]: @felixschorer would you be able to add an integration test case for `join` API?

**Pull request merged**